mypy/checker.py

```diff
@@ -2482,16 +2525,9 @@ def find_isinstance_check(node: Node,
             type_map,
             weak,
         )
-        if left_if_vars:
-            if right_if_vars is not None:
-                left_if_vars.update(right_if_vars)
-            else:
-                left_if_vars = None
-        else:
-            left_if_vars = right_if_vars
 
-        # Make no claim about the types in else
-        return left_if_vars, {}
+        return (or_conditional_maps(left_if_vars, right_if_vars),
```

Comment:
Again, it would be good to describe this. Perhaps just refer to the 'and' case and discuss how this is different.